lib/smart_proxy_openscap/openscap_api.rb

```diff
@@ -42,5 +43,16 @@ class Api < ::Sinatra::Base
 
       {"created" => true}.to_json
     end
+
+    get "/policies/:policy_id/content" do
+      content_type 'application/xml'
+      begin
+        Proxy::OpenSCAP::get_policy_content(params[:policy_id])
+      rescue OpenSCAPException => e
+        log_halt e.http_code, "Error fetching xml file: #{e.message}"
+      rescue StandardError => e
+        log_halt 500, "Error occurred: #{e.message}"
+      end
+    end
   end
 end
```

Comment:
As an aside, there's a helper to verify client cert: https://github.com/theforeman/smart-proxy/blob/develop/lib/sinatra/authorization.rb#L29. But you'd need to break up the api into independent modules to use it (as not all calls require the check). Not sure if it's worth it atm.